extensions/ql-vscode/src/cli.ts

```diff
@@ -751,7 +751,7 @@ export class CodeQLCliServer implements Disposable {
       const dot = await this.readDotFiles(interpretedResultsPath);
       return dot;
     } catch (err) {
-      throw new Error(`Reading output of interpretation failed: ${err.stderr || err}`);
+      throw new Error(`Reading output of interpretation failed: ${getErrorMessage(err)}`);
```

Comment:
I think that was an error. There is no `stderr` property on the Error class.

Reply:
There were other places in this file where `err.stderr || err` is translated to
`(err as any).stderr || getErrorMessage(err)`.  Is the difference due to the `catch`  blocks catching different objects?  If so, it might be useful to document how we can tell which expression to use where, for the next person who needs to modify this file.

Reply:
That's interesting. I am really struggling to find out why `err.stderr` is being used, or if that property even exists.

I suspect that it is coming from here:

https://github.com/github/vscode-codeql/blob/cf99c5e79c8cddd2a56e9f047b3670a5a035efbe/extensions/ql-vscode/src/cli.ts#L1048

```
await promisify(child_process.execFile)(...);
```

returns an object with `{ stderr, stdout }`. It may be that when catching the error from a child process, the `stderr` property is being read (even though that's a mistake and it doesn't exist). The other locations in the code could just be copies of the first.

Reply:
I think you have a better handle on what is happening here than I do, so I will let you do what you think is best.